File: python/safe_calculator.py
class SafeCalculator:
    def __init__(self, authorizer):
        self.authorizer = authorizer

    def add(self, left, right):
        authorized = self.authorizer.authorize()
        # Bug! Should be `if not authorized`
        if authorized:
            raise Exception("Not authorized")
        return left + right

File: ts/safe-calculator.ts
export interface Authorizer {
  authorize(): boolean
}


export class SafeCalculator {
  authorizer: Authorizer

  constructor(authorizer: Authorizer) {
    this.authorizer = authorizer
  }

  add(left: number, right: number) {
    const authorized = this.authorizer.authorize()
    // Bug! Should be `if(!authorized)`
    if (authorized) {
      throw new Error('Not authorized')
    }
    return left + right
  }
}

